GitCommands/Git/GitGpgController.cs

```diff
@@ -200,14 +187,44 @@ public string GetCommitVerificationMessage()
             return _module.RunGitCmd($"log --pretty=\"format:%GG\" -1 {_revision.Guid}");
         }
 
-        private string GetTagVerificationMessage(string tag, bool raw = true)
+        private string GetTagVerificationMessage(IGitRef tagRef, bool raw = true)
         {
-            if (string.IsNullOrWhiteSpace(tag))
+            string tagName = tagRef.LocalName;
+            if (string.IsNullOrWhiteSpace(tagName))
                 return null;
 
             string rawFlag = raw == true ? "--raw" : "";
 
-            return _module.RunGitCmd($"verify-tag {rawFlag} {tag}");
+            return _module.RunGitCmd($"verify-tag {rawFlag} {tagName}");
+        }
+
+        public string GetTagVerifyMessage()
+        {
+            return _tagVerifyMessage.Value;
+        }
+
+        private string EvaluateTagVerifyMessage()
+        {
+            if (_usefulTagRefs.Count == 0)
+            {
+                return "";
+            }
+            else if (_usefulTagRefs.Count == 1)
+            {
+                return GetTagVerificationMessage(_usefulTagRefs[0], false);
+            }
+            else
+            {
+                string tagVerifyMessage = "";
+
+                /* Only to populate TagVerifyMessage */
+                foreach (var tagRef in _usefulTagRefs)
+                {
+                    /* String printed in dialog box */
+                    tagVerifyMessage = $"{tagVerifyMessage}{tagRef.LocalName}\r\n{GetTagVerificationMessage(tagRef, false)}\r\n\r\n";
+                }
+                return tagVerifyMessage;
+            }
```

Comment:
you can reduce nesting by removing unneded "else"s since each condition returns

Reply:
I know that.

Reply:
I'll try to merge this PR and also the #6 on my branch to have something working. thanks.